### PR TITLE
Add BLQ configuration surface to PIConfiguration

### LINKED ISSUE #248: Add `blqMethod` and `blqOptions` fields to `PIConfiguration`
`objectiveFunctionType = "m3"` conflates observation scoring with BLQ handling, forcing M3 parameters (`linScaleCV`, `logScaleSD`) into `objectiveFunctionOptions` and requiring `scaling` to be secretly injected at runtime from `PIOutputMapping`.

Add `blqMethod` (`"none"` | `"lloq_half"` | `"m3"`), `blqRemove`, and `blqOptions` to `PIConfiguration`. Remove `"m3"` from `objectiveFunctionType` and move M3 parameters out of `ObjectiveFunctionSpecs`.

### PR DESCRIPTION
**Pull request by rengelke**
Right now, one setting (`objectiveFunctionType = "m3"`) tries to answer two unrelated questions at once: how the model fit is scored, and how measurements below the detection limit (BLQ) are treated. This PR gives each question its own setting, which removes the awkward workarounds that the combined setting forced, like BLQ options ending up in the wrong place and the scaling having to be patched in the fitting loop.

Separates below-limit-of-quantification (BLQ) handling from the scoring objective, which were previously conflated in `objectiveFunctionType`. This is a configuration-surface change only, and existing least-squares results are numerically unchanged.

- Add `blqMethod`, `blqRemove`, and `blqOptions` fields to `PIConfiguration` with validation in the active-binding setters, backed by the new `BLQMethods`, `BLQRemoveModes`, and `BLQOptions` enums and `BLQOptionSpecs`.
- Remove `"m3"` from `objectiveFunctionType` and relocate `linScaleCV` and `logScaleSD` from `objectiveFunctionOptions` into `blqOptions`.

Closes #248

## Summary by CodeRabbit

* **New Features**
  * Added BLQ configuration to parameter identification via `blqMethod`, `blqRemove`, and `blqOptions`, including new help topics for BLQ methods/options and configuration bindings.
  * `blqRemove` is validated and stored, but BLQ removal is not applied yet (future release).

* **Bug Fixes**
  * BLQ/censored handling in objective and cost metrics is now driven by `blqMethod` (with `m3` selecting censored-likelihood behavior).
  * Least-squares results remain numerically unchanged.

* **Documentation**
  * Updated user guidance and vignettes to reflect the new BLQ configuration.

### CODE REVIEW

File: R/enums.R [hunk 1]
```diff
@@ -191,26 +191,21 @@ CIDefaults <- list(
...
+#' - **`scaleVar`** - Whether residual scaling is applied. Default is `FALSE`.
 #'
 #'   These options are configurable in `PIConfiguration`, directly influencing
 #'   the `calculateCostMetrics` functionality for detailed model fit assessment.
 ObjectiveFunctionOptions <- ospsuite.utils::enum(list(
   objectiveFunctionType = "lsq",
   residualWeightingMethod = "none",
   robustMethod = "none",
-  scaleVar = FALSE,
-  linScaleCV = 0.2,
```
Comment (PavelBal) [inline R/enums.R:212]: These options have been removed? Why? Mention it in NEWS, explain the decision to remove.

File: R/enums.R [hunk 2]
```diff
@@ -221,14 +216,78 @@ ObjectiveFunctionOptions <- ospsuite.utils::enum(list(
...
+#' BLQ Removal Modes
+#'
+#' Modes selecting which below-limit-of-quantification (BLQ) observations enter
+#' the objective function.
+#'
+#' @export
+#' @name BLQRemoveModes
+#' @details The available modes are:
+#' - **`none`** - Keep all BLQ observations (default).
+#' - **`always`** - Remove all BLQ observations (Beal M1, PK-Sim "Always").
```
Comment (PavelBal) [inline R/enums.R:236]: Explain what Beal M1, M6 means. If this described in the vignette - refer to the it.

File: R/enums.R [hunk 3]
```diff
@@ -221,14 +216,78 @@ ObjectiveFunctionOptions <- ospsuite.utils::enum(list(
...
+#' - **`trailingSingle`** - Keep the first point of each trailing BLQ run per
+#'   dataset and remove the rest (Beal M6, PK-Sim "Reduce trailing").
+#'
+#' Note: in the current release `blqRemove` is validated and stored but not yet
```
Comment (PavelBal) [inline R/enums.R:240]: Don't understand this. If this does not work yet - do not expose it.

File: R/enums.R [hunk 4]
```diff
@@ -221,14 +216,78 @@ ObjectiveFunctionOptions <- ospsuite.utils::enum(list(
...
+#' @export
+#' @name BLQMethods
+#' @details The available methods are:
+#' - **`none`** - Use the BLQ value as stored, with no special handling.
+#' - **`lloq`** - Clamp both observed and simulated values to the LLOQ (PK-Sim
+#'   default).
+#' - **`lloqHalf`** - Substitute the observed value with half the LLOQ (Beal M5).
+#' - **`m3`** - Add the censored-likelihood contribution (Beal M3).
+#'
+#' Note: in the current release `none`, `lloq`, and `lloqHalf` are not yet
```
Comment (PavelBal) [inline R/enums.R:264]: Same here - if it does not work yet, don't expose.

File: R/enums.R [hunk 5]
```diff
@@ -221,14 +216,78 @@ ObjectiveFunctionOptions <- ospsuite.utils::enum(list(
...
+#' `blqOptions` field of `PIConfiguration`.
+#'
+#' @export
+#' @name BLQOptions
+#' @details Settings include:
+#' - **`linScaleCV`** - Coefficient of variation for linear scaling, applied to
+#'   `lloq` values. Default is `0.2`.
+#' - **`logScaleSD`** - Standard deviation for logarithmic scaling. Default is
+#'   derived from a coefficient of variation of `0.2`.
+BLQOptions <- ospsuite.utils::enum(list(
```
Comment (PavelBal) [inline R/enums.R:290]: Ah, they have been moved here. Worth mentioning in NEWS.

File: vignettes/error-calculation.Rmd [hunk 6]
```diff
@@ -20,31 +20,46 @@ knitr::opts_knit$set(
...
-The `lsq` method is a traditional approach that minimizes the sum of the squared differences between observed and simulated values. This method is widely used in pharmacokinetic modeling, offering a straightforward and intuitive error calculation. By default, selecting this method will transform BQL data by replacing it with LLOQ/2.
+The error model scores how far the simulation is from the observed data.
 
-### M3 - Extended Least Squares Error
-
-The `m3` method extends the least squares error by incorporating the likelihood of observed data being below the quantification limit. This method treats BQL data as censored, applying maximum likelihood estimation to include all data in the model fit, thus offering a more accurate and statistically robust approach to handling BQL observations.
+### LSQ - Least Squares Error
 
-### Configuring Error Models in PI
+The `lsq` method minimizes the sum of the squared differences between observed and simulated values. It is a widely used, straightforward, and intuitive error calculation for pharmacokinetic modeling, and is currently the only available `objectiveFunctionType`.
```
Comment (PavelBal) [inline vignettes/error-calculation.Rmd:32]: Add an equation showing how the error is calculated.

File: vignettes/error-calculation.Rmd [hunk 7]
```diff
@@ -20,31 +20,46 @@ knitr::opts_knit$set(
...
+
+`blqRemove` selects which BLQ observations enter the fit at all: `none` (default) keeps them all, `always` drops them all, and `trailingSingle` keeps only the first point of each trailing BLQ run.
+
+```{r, eval=FALSE}
+piConfiguration$blqMethod <- "m3"
+piConfiguration$blqOptions <- list(linScaleCV = 0.2)
 ```
 
-The error calculation between simulated results and observer datasets within each `PIOutputMapping` directly affects the optimization process, aggregating errors across all mappings.
+In the current release the substitution methods (`none`, `lloq`, `lloqHalf`) share the same half-LLOQ behaviour, and `blqRemove` is validated but not yet applied during optimization; these distinctions are introduced in later releases.
```
Comment (PavelBal) [inline vignettes/error-calculation.Rmd:79]: Same here - don't expose something that is not implemented.

File: vignettes/error-calculation.Rmd [hunk 8]
```diff
@@ -20,31 +20,46 @@ knitr::opts_knit$set(
...
+## Handling Data Below the Quantification Limit (BLQ)
+
+A BLQ measurement is known only to lie between zero and the LLOQ. Two `PIConfiguration` settings control how such observations are treated.
+
+`blqMethod` selects how a retained BLQ observation contributes to the error:
+
+- **`lloqHalf`** (default): substitute the BLQ value with half the LLOQ.
+- **`lloq`**: clamp the BLQ value to the LLOQ.
+- **`none`**: use the stored BLQ value unchanged.
+- **`m3`**: treat the BLQ value as censored and score it by the likelihood that the prediction falls below the LLOQ (Beal's M3 method), keeping all data in the fit for a more statistically robust result. The `blqOptions` field (`linScaleCV`, `logScaleSD`) parameterises this censored contribution.
```
Comment (PavelBal) [inline vignettes/error-calculation.Rmd:61]: Explain what the "Beals" method is, or refer to the publication.

**Review by PavelBal — CHANGES_REQUESTED** [review]

**Review by nautilus69 — APPROVED** [review]
With minor consideration because Pavel comments are the main concerns. 

1. Add a runtime warning/message when the user selects a BLQ setting that isn't yet differentiated at runtime, so the docs' "not yet applied" caveat is enforced in behavior and not just prose. Two cases:
    - blqMethod set to "none" or "lloq" (both currently behave like "lloqHalf").
    - blqRemove set to "always" or "trailingSingle" (both currently remove nothing).

  2. Give the objectiveFunctionType = "m3" rejection a migration-specific message pointing to blqMethod = "m3".

  4. Add regression tests that lock the current documented-but-inert behavior: blqMethod = "none" and "lloqHalf" produce the same cost, and blqRemove = "always"/"trailingSingle" leaves the fit identical to "none".

File: R/enums.R (hunk 3 above)
Comment (rengelke) [reply R/enums.R:240]: Right, my merging plan was not obvious. Targeting integration branch `mle-objective-and-blq-handling` - so this is the initial part of a bigger implementation: MLE path and BLQ options

File: R/enums.R (hunk 1 above)
Comment (rengelke) [reply R/enums.R:212]: Moved; NEWS entry exists

File: R/enums.R (hunk 5 above)
Comment (rengelke) [reply R/enums.R:290]: Moved, NEWS updated

File: vignettes/error-calculation.Rmd (hunk 7 above)
Comment (rengelke) [reply vignettes/error-calculation.Rmd:79]: Defer to stack

File: vignettes/error-calculation.Rmd (hunk 6 above)
Comment (rengelke) [reply vignettes/error-calculation.Rmd:32]: Added https://github.com/Open-Systems-Pharmacology/OSPSuite.ParameterIdentification/pull/278/commits/deff49f5419ffc03d851c99cd34711f94ecfd0f7, more changes in vignette planned later in the stack

**Comment by rengelke** [comment]
> 1. Add a runtime warning/message when the user selects a BLQ setting that isn't yet differentiated at runtime, so the docs' "not yet applied" caveat is enforced in behavior and not just prose. Two cases:
>    
>    * blqMethod set to "none" or "lloq" (both currently behave like "lloqHalf").
>    * blqRemove set to "always" or "trailingSingle" (both currently remove nothing).
> 2. Give the objectiveFunctionType = "m3" rejection a migration-specific message pointing to blqMethod = "m3".
> 3. Add regression tests that lock the current documented-but-inert behavior: blqMethod = "none" and "lloqHalf" produce the same cost, and blqRemove = "always"/"trailingSingle" leaves the fit identical to "none".

This is first of 6 PRs for the MLE/BLQ rework. So not addressing here as it will be either resolved or dropped entirely later.

File: vignettes/error-calculation.Rmd (hunk 8 above)
Comment (rengelke) [reply vignettes/error-calculation.Rmd:61]: Added in https://github.com/Open-Systems-Pharmacology/OSPSuite.ParameterIdentification/pull/278/commits/3d4d7f4f626992f4ba27baac5c3bf8c2d0b6e20f

File: R/enums.R (hunk 2 above)
Comment (rengelke) [reply R/enums.R:236]: Added in https://github.com/Open-Systems-Pharmacology/OSPSuite.ParameterIdentification/pull/278/commits/3d4d7f4f626992f4ba27baac5c3bf8c2d0b6e20f

**Review by PavelBal — APPROVED** [review]
